ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -1,68 +1,88 @@
     useEffect(() => {
-        if (!data || !data.results) {
-            setAvailableNamespaces([]);
-        } else {
-            setAvailableNamespaces(data.results);
-        }
-        // clean up state on unmount
-        return () => setAvailableNamespaces([]);
-    }, [data]);
+        const namespaceIds = JSON.parse(selectedNamespaceIdsAsString);
+        if (namespaceIds.length > 0) {
+            setDeploymentResponse({
+                loading: true,
+                error: '',
+                deploymentsByNamespace: [],
+            });
+            const searchQuery: Record<string, string[]> = {
+                'Namespace ID': namespaceIds,
+            };
+            const sortOption = { field: 'Deployment', reversed: 'false' };
+            listDeployments(searchQuery, sortOption, 0, 0)
+                .then((response) => {
+                    const namespacesWithDeployments: NamespaceWithDeployments[] = [];
+                    const deploymentsByNamespace = new Map<string, Deployment[]>();
+                    response.forEach(({ id, name, namespace }) => {
+                        const deployment = { id, name };
+                        const deploymentList = deploymentsByNamespace.get(namespace);
+                        if (deploymentList) {
+                            deploymentList.push(deployment);
+                            deploymentsByNamespace.set(namespace, deploymentList);
+                        } else {
+                            deploymentsByNamespace.set(namespace, [deployment]);
+                        }
+                    });
+                    deploymentsByNamespace.forEach((deployments, namespaceName) => {
+                        const namespaceWithDeployments: NamespaceWithDeployments = {
+                            metadata: {
+                                name: namespaceName,
+                            },
+                            deployments,
+                        };
+                        namespacesWithDeployments.push(namespaceWithDeployments);
+                    });
+                    setDeploymentResponse({
+                        loading: false,
+                        error: '',
+                        deploymentsByNamespace: namespacesWithDeployments,
+                    });
+                })
+                .catch((error) => {
+                    const message = getAxiosErrorMessage(error);
+                    const errorMessage =
+                        message ||
+                        'An unknown error occurred while getting the list of deployments';
 
-    return {
-        loading,
-        error,
-        deploymentsByNamespace: availableDeployments,
-    };
+                    setDeploymentResponse({
+                        loading: false,
+                        error: errorMessage,
+                        deploymentsByNamespace: [],
+                    });
+                });
+        }
+    }, [selectedNamespaceIdsAsString]);
```

Comment:
React doc ( https://react.dev/reference/react/useEffect ) states: 
```
React will compare each dependency with its previous value using the [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison.
```
Comparing strings is then a deep compare while array comparison is not. To work around array comparison triggering infinite effect loop, the array is serialized for the effect dependency comparison to compare string contents, then de-serialized for the effect to use the actual array content.

Reply:
Why not use `useDeepCompareEffect` here?